st_package_reviewer/check/file/__init__.py

```diff
@@ -1,13 +1,14 @@
 import functools
 import itertools
 from pathlib import Path
+import ast
 
 from .. import Checker, find_all
 
 __all__ = ('FileChecker', 'get_checkers')
 
 
-class FileChecker(Checker):
+class FileChecker(Checker, ast.NodeVisitor):
```

Comment:
I thought of using a separate `ASTChecker` class that would implement utility functions only relevant for checkers that rely on the AST (and thus only work on Python files). You should move the AST checkers into a sub-package of the file package, too.

I'm okay with this class also subclassing `ast.NodeVisitor` though, which should make for more condense code.